src/write_gcm.c

```diff
@@ -3737,12 +3872,60 @@ static int wg_rebase_item(c_avl_tree_t *deriv_tree, wg_payload_t *payload,
   return 0;
 }
 
+static c_avl_tree_t *wg_group_payloads_by_host(wg_payload_t *head, int *count) {
+  c_avl_tree_t *host_tree = c_avl_create((
+      int (*)(const void*, const void*))&strcmp);
+  if (host_tree == NULL) return NULL;
+  int host_count = 0;
+  while (head != NULL) {
+    const char *host = head->key.host;
+    wg_payload_hook_t *hook = NULL;
```

Comment:
Is there any need to initialize |hook|? Seems like it gets assigned below no matter what.

Same question for |value|, though this one is more subtle because it gets set in c_avl_get().